### PR TITLE
[LIQ-656] improve filter LIQNFT

### PR DESCRIPTION
**Pull request by vanminh1701**
uncomment

### CODE REVIEW

**Comment by haVincy** [comment]
Test link - https://core-improve-filter-orders.d22tqykss69onq.amplifyapp.com/

File: core/ui/src/components/ShopFilter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Search } from 'components/Search';
+import { CandyShop, fetchAllShop } from '@liqnft/candy-shop-sdk';
+import { LoadStatus } from 'constant';
+import { ListBase, CandyShop as CandyShopResponse } from '@liqnft/candy-shop-types';
+import { Processing } from 'components/Processing';
+import { ShopFilter as ShopFilterType } from 'model';
```
Comment (haVincy) [inline core/ui/src/components/ShopFilter/index.tsx:7]: `ShopFilterInfo` ?

File: core/ui/src/components/ShopFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            const duplicateList = [...list, ...result];
+            const newList: CandyShopResponse[] = [];
+            const memo: any = {};
+
+            duplicateList.forEach((item) => {
+              if (memo[item.candyShopAddress]) return;
+              newList.push(item);
+              memo[item.candyShopAddress] = true;
+            });
+            return newList;
```
Comment (haVincy) [inline core/ui/src/components/ShopFilter/index.tsx:74]: These part seems can be extract common as utils for both `ShopFilter` and `CollectionFilter` or maybe more filters in future ? Not sure how should do it, just a feelings, maybe hook can serve that? What's your thought.

File: core/ui/src/components/Search/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { IconSearch } from 'assets/IconSearch';
+import './index.less';
+
+interface SearchProps {
+  onSearch: (search: string) => void;
+  placeholder?: string;
+}
+
+const DEBOUNCE_TIME = 300;
+export const Search: React.FC<SearchProps> = ({ onSearch, placeholder = '' }) => {
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/components/Search/index.tsx:11]: yes, great idea. But I think I should keep it as current. If we meet this case in future, will update it or maybe refactor it in a seperate task

File: core/ui/src/components/CollectionFilter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  }, [fetchOption, filters, offset]);
+
+  // Manual filter
+  const filteredList: CollectionFilterType[] = useMemo(() => {
+    if (!Array.isArray(filters)) return [];
+    if (!keyword) return filters;
+
+    const keywordList = keyword.split(' ');
+    return filters.filter((item) => {
+      const name = (item.name as string).toLowerCase();
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/components/CollectionFilter/index.tsx:95]: yes, guaranteed

File: core/ui/src/utils/removeDuplicateFromList.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  const duplicateList = [...oldList, ...addList];
+  const newList: T[] = [];
+  const memo: any = {};
+  for (const item of duplicateList) {
+    if (memo[item[key]]) break;
+    newList.push(item);
+    memo[item[key]] = true;
+  }
+  return newList;
+}
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/ui/src/utils/removeDuplicateFromList.ts:11]: I will rename it to `array.ts`

File: core/sdk/src/factory/backend/ShopAPI.ts [hunk 6]
```diff
@@ -26,3 +49,42 @@ export async function fetchShopStatusByShopId(
...
+  }
+  const queryString = qs.stringify(queryObj);
+  return await axiosInstance
+    .get<ListBase<NftCollection>>(`/nftCollection?${queryString}`)
+    .then((response) => response.data);
+}
+
+export async function fetchCollectionByShopId(
+  axiosInstance: AxiosInstance,
+  queryDto?: CollectionQuery
```
[the remark this answers is not in the document]
Comment (vanminh1701) [reply core/sdk/src/factory/backend/ShopAPI.ts:74]: `DTO stands for data transfer object. As the name suggests, a DTO is an object made to transfer data.`
I think we had typescript to check what is inside that query object. So don't need to strict on it. It likes a general name which can be accepted

[the remark this answers is not in the document]
Comment (haVincy) [reply core/sdk/src/factory/backend/ShopAPI.ts]: okay... it's my first time to learn about this. 
For Typescript case here, the  `interface` is more like the Data Modals not the Data Object but the core concept are the same. 

If we want to formalize data between client <-> server, it seems we should make the DTO in core/types?

**Review by haVincy — APPROVED** [review]

**Pull request merged**